Add ikos-underlay and ikos builds.

nuclearsandwich opened the pull request:
The ikos-underlay job builds [IKOS] (currently from a custom fork) in an
underlay by itself.

This is done so that its compiler wrappers can be used during the build
of a ROS workspace in order to facilitate analysis with the nascent
[ament_ikos] package.

[IKOS]: https://github.com/NASA-SW-VnV/ikos
[ament_ikos]: https://github.com/ament/ament_ikos

nuclearsandwich committed:
Add ikos-underlay and ikos builds.

The ikos-underlay job builds [IKOS] (currently from a custom fork) in an
underlay by itself.

This is done so that its compiler wrappers can be used during the build
of a ROS workspace in order to facilitate analysis with the nascent
[ament_ikos] package.

[IKOS]: https://github.com/NASA-SW-VnV/ikos
[ament_ikos]: https://github.com/ament/ament_ikos

mjeronimo reviewed (approved):
I'm not familiar with the details of the YAML file for CI, but LGTM.

mjeronimo reviewed (approved):
I'm not familiar with the details of the YAML file for CI, but LGTM.

nuclearsandwich committed:
Add ikos-underlay dependencies to ikos build.

Not all of these are strictly runtime dependencies but in trying to get
the build running I've just brought all of them forward.

nuclearsandwich committed:
Use repos file from default spaceros branch.

nuclearsandwich committed:
Ignore cobra packages which don't build under clang.

nuclearsandwich committed:
Update repos file url.

Update repos file urls.

nuclearsandwich committed:
Skip rosidl_typesupport_fastrtps_cpp as well as others.

clalancette commented:
Since we ended up removing these jobs, I'm going to close this PR.  I don't think we are going to move forward with this.  Just in case, I won't delete the branch for now.
